### PR TITLE
feat: [SRM-12801] Update Error Tracking notification templates and data

### PR DESCRIPTION
**Pull request by ricksjames**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)

### CODE REVIEW

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java [hunk 1]
```diff
@@ -1925,19 +1937,35 @@ NotificationData getNotificationData(
...
-            templateDataMap.put(ERROR_TRACKING_TYPE, "<INSERT CODE ERROR DATA>");
+            return NotificationData.builder().shouldSendNotification(false).build();
           }
-          // Always send a notification for this current iteration as the event summary api isn't expected to work
-          // initially
+          templateDataMap.put(
+              NOTIFICATION_URL, buildMonitoredServiceConfigurationTabUrl(baseLinkUrl, monitoredServiceParams));
+          templateDataMap.put(NOTIFICATION_NAME, notificationName);
+          templateDataMap.put(ENVIRONMENT_NAME, environmentId);
           return NotificationData.builder().shouldSendNotification(true).templateDataMap(templateDataMap).build();
```
Comment (cseverino789) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java:1969]: This Case statement has too many return statements that can be avoided should reduce to just two like the case above.

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java [hunk 2]
```diff
@@ -1925,19 +1937,35 @@ NotificationData getNotificationData(
         if (featureFlagService.isFeatureFlagEnabled(SRM_CODE_ERROR_NOTIFICATIONS, monitoredService.getAccountId())) {
           log.info("SRM_CODE_ERROR_NOTIFICATIONS feature flag enabled");
```
Comment (cseverino789) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java:1938]: Can this log be removed?

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java [hunk 3]
```diff
@@ -1946,6 +1974,78 @@ NotificationData getNotificationData(
     }
   }
 
+  private Map<String, String> getCodeErrorNotificationData(
```
Comment (cseverino789) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java:1977]: This is a lot of Code Errors / ET specific code being added to the over-arching MontiorService class. Can these methods be moved to another ET specific class?

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java [hunk 4]
```diff
@@ -6,37 +6,111 @@
...
+
+  public static final String EMAIL_LINK_BEGIN = "<a style=\"text-decoration: none; color: #0278D5;\" href=\"";
+  public static final String EMAIL_LINK_MIDDLE = "\">";
+  public static final String EMAIL_LINK_END = "</a>";
+
   @Override
-  protected String getHeaderMessage(Map<String, String> notificationDataMap) {
-    return "observed new events for the following event summaries: " + notificationDataMap.get(ERROR_TRACKING_TYPE)
-        + " for";
+  public Map<String, String> getTemplateData(ProjectParams projectParams, String name, String identifier,
```
Comment (cseverino789) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java:49]: Recommend adding Unit test for this class if possible.

**Review by cseverino789 — CHANGES_REQUESTED** [review]

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java (hunk 1 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java:1969]: will fix

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java (hunk 2 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java:1938]: yes, will fix

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java (hunk 3 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java:1977]: yes, agree, will fix.

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java (hunk 4 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java:49]: yes, agree, will fix.

**Comment by ricksjames** [comment]
trigger ti2

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java (hunk 1 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java]: done

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java (hunk 2 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java]: done

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java (hunk 3 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java]: done

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java (hunk 4 above)
Comment (ricksjames) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java:49]: done

**Review by cseverino789 — APPROVED** [review]

**Review by cseverino789 — APPROVED** [review]
> Reviewed 9 of 12 files at r1, 1 of 2 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: :shipit: complete! ✓

**Review by praveensugavanam — CHANGES_REQUESTED** [review]
> Reviewed 6 of 12 files at r1, 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: 1 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NL74BCDEX3oqtORxgPo:b8272o6) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
> ```Java
>   @VisibleForTesting
>   NotificationData getNotificationData(MonitoredService monitoredService,
>       MonitoredServiceNotificationRuleCondition condition, String notificationUuid, String notificationName) {
> ```

Why are we adding these params to the generic getNotificationData method ? How are the other cases working without these params and why do we need them for ET ?
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74RqsDbxZyDnuLYGq:-NL74RqsDbxZyDnuLYGr:b-ao1c62) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java#L26)):*
> ```Java
> public class ErrorTrackingTemplateDataGenerator
>     extends MonitoredServiceTemplateDataGenerator<MonitoredServiceCodeErrorCondition> {
>   // Variables from the Monitored Service condition that triggered the notification
> ```

a lot of these seem like they could be common and not ET specific, Can you add them to a common constants file for notifications ? 
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/utils/ErrorTrackingNotificationRuleUtils.java` line 50 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74kVF9uvdw5h5n1iS:-NL74kVF9uvdw5h5n1iT:b-mfbc57) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/utils/ErrorTrackingNotificationRuleUtils.java#L50)):*
> ```Java
>   }
> 
>   public static List<ErrorTrackingEvent> getErrorTrackingEventsRecursive(
> ```

im not clear about the logic here. Please get it reviewed by the ET team as well.

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NL7684cDS7SjgCAzmqQ:b-6zni8t) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, praveensugavanam (Praveen Kambam Sugavanam) wrote…</i></summary><blockquote>

Why are we adding these params to the generic getNotificationData method ? How are the other cases working without these params and why do we need them for ET ?
</blockquote></details>

The notificationName is needed by ET as it is specified in the UX notification template design to be displayed as part of the notification. 

The notificationUuid is used by error tracking to keep track of which events have been notified for which notification. For example, if two notifications with the exact same settings are setup, one for slack and one for email, we need to ensure that both email and slack get notified with all the expected notifications. Without the notification Uuid  being sent to the ET backend, this management would need to be done on the CV side. We determined it would be better for the ET backend side to track this, but need the CV notificationUuid unique identifier to keep track of which notification is making the request, hence the addition.

I believe this is not needed by the existing CV notifications due to the fact that those notifications are only dealing with a single event, while ET notifications notify of a group of events based on settings and a time frame.
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74RqsDbxZyDnuLYGq:-NL7C0Wy3BWtX_rWuVLR:b-9r86z2) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java#L26)):*
<details><summary><i>Previously, praveensugavanam (Praveen Kambam Sugavanam) wrote…</i></summary><blockquote>

a lot of these seem like they could be common and not ET specific, Can you add them to a common constants file for notifications ? 
</blockquote></details>

Will do, moving the following more generic values that may be used outside ET in the future to NotificationRuleConstants.java:
ENVIRONMENT_NAME
MONITORED_SERVICE_URL
NOTIFICATION_URL
NOTIFICATION_NAME

**Review by abeg777 — APPROVED** [review]

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/utils/ErrorTrackingNotificationRuleUtils.java` line 50 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74kVF9uvdw5h5n1iS:-NL7q5-I9wIpxvfdlP4c:b-ndpfdk) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/utils/ErrorTrackingNotificationRuleUtils.java#L50)):*
<details><summary><i>Previously, praveensugavanam (Praveen Kambam Sugavanam) wrote…</i></summary><blockquote>

im not clear about the logic here. Please get it reviewed by the ET team as well.
</blockquote></details>

Done. Had Abraham Gutierrez review in addition to Corey Severino's review.

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: 5 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74RqsDbxZyDnuLYGq:-NL86bdR-YwRzFzwDD9m:b-5zuv1l) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/notification/services/impl/ErrorTrackingTemplateDataGenerator.java#L26)):*
<details><summary><i>Previously, ricksjames (James Ricks) wrote…</i></summary><blockquote>

Will do, moving the following more generic values that may be used outside ET in the future to NotificationRuleConstants.java:
ENVIRONMENT_NAME
MONITORED_SERVICE_URL
NOTIFICATION_URL
NOTIFICATION_NAME
</blockquote></details>

Done, changing the above turns out to cause an obscure build issue which seems related to lombok experimental features being used in the baseline. None of the new common constants are duplicated, nor are they used by other notification types. For now I would suggest not moving these, and in the future if any are needed in a common constant file move them then.

**Comment by praveensugavanam** [comment]
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NLO1rAyEM29DfBvuwe6:b581540) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, ricksjames (James Ricks) wrote…</i></summary><blockquote>

The notificationName is needed by ET as it is specified in the UX notification template design to be displayed as part of the notification. 

The notificationUuid is used by error tracking to keep track of which events have been notified for which notification. For example, if two notifications with the exact same settings are setup, one for slack and one for email, we need to ensure that both email and slack get notified with all the expected notifications. Without the notification Uuid  being sent to the ET backend, this management would need to be done on the CV side. We determined it would be better for the ET backend side to track this, but need the CV notificationUuid unique identifier to keep track of which notification is making the request, hence the addition.

I believe this is not needed by the existing CV notifications due to the fact that those notifications are only dealing with a single event, while ET notifications notify of a group of events based on settings and a time frame.
</blockquote></details>

If they are just needed for ET, we should write an overloaded method with more params so you can use that for ET. Forcing the same params for others doesnt make sense IMO. 

You can add an overloaded method and refactor the common logic underneath so that you dont reuse repeatable code.

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: Some of the checks are too old. 5 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NLR2WiFEpvpUr7ag2Gc:bfrb0x0) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, praveensugavanam (Praveen Kambam Sugavanam) wrote…</i></summary><blockquote>

If they are just needed for ET, we should write an overloaded method with more params so you can use that for ET. Forcing the same params for others doesnt make sense IMO. 

You can add an overloaded method and refactor the common logic underneath so that you dont reuse repeatable code.
</blockquote></details>

Will do, agree

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: Some of the checks are too old. 5 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NLRBJNeAKEXQDIKu3Vg:b-jbye5g) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, ricksjames (James Ricks) wrote…</i></summary><blockquote>

Will do, agree
</blockquote></details>

Upon further review this doesn't fit well with the current design and would require additional refactoring of existing logic to implement.

The switch statement which determines the notification condition type is inside the MonitoredServiceServiceImpl.getNotificationData(...) method. 

    switch (condition.getType()) {
      case HEALTH_SCORE:
       ...
      case CODE_ERRORS:
       ...

Calling an overloaded version of it depending on condition type would require pulling the switch statement out of MonitoredServiceServiceImpl.getNotificationData(...) and moving it up into MonitoredServiceServiceImpl.handleNotification(...). 

Please advise.

**Comment by praveensugavanam** [comment]
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NLY3mWAB4uwabzi01V4:bmbyfu6) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, ricksjames (James Ricks) wrote…</i></summary><blockquote>

Upon further review this doesn't fit well with the current design and would require additional refactoring of existing logic to implement.

The switch statement which determines the notification condition type is inside the MonitoredServiceServiceImpl.getNotificationData(...) method. 

    switch (condition.getType()) {
      case HEALTH_SCORE:
       ...
      case CODE_ERRORS:
       ...

Calling an overloaded version of it depending on condition type would require pulling the switch statement out of MonitoredServiceServiceImpl.getNotificationData(...) and moving it up into MonitoredServiceServiceImpl.handleNotification(...). 

Please advise.
</blockquote></details>

I think a switch statement is better than having params that arent required for processing in all paths. So let's go with that.

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: Some of the checks are too old. 5 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NLaVu9Y7eX38PZEFJIO:bhy9ot6) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, praveensugavanam (Praveen Kambam Sugavanam) wrote…</i></summary><blockquote>

I think a switch statement is better than having params that arent required for processing in all paths. So let's go with that. 
</blockquote></details>

Will do.

**Review by ricksjames — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: 7 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java` line 1858 at r4](https://reviewable.io/reviews/harness/harness-core/41706#-NL74BCDEX3oqtORxgPn:-NLgEfW078Gwa8iwaD4K:b-896fjx) ([raw file](https://github.com/harness/harness-core/blob/8a60dedec68692d084d22c2b8c1898cbe3e3dc1a/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/monitoredService/MonitoredServiceServiceImpl.java#L1858)):*
<details><summary><i>Previously, ricksjames (James Ricks) wrote…</i></summary><blockquote>

Will do.
</blockquote></details>

Done

**Review by cseverino789 — APPROVED** [review]
> Reviewed 2 of 3 files at r6, 5 of 5 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: 1 change requests. 1. unresolved discussions.

**Review by praveensugavanam — APPROVED** [review]
> Reviewed 2 of 12 files at r1, 1 of 4 files at r3, 2 of 3 files at r6, 5 of 5 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41706)* status: :shipit: complete! ✓

**Review by abeg777 — APPROVED** [review]

**Pull request merged**